cpp/scenario/gazebo/include/scenario/gazebo/Model.h

```diff
@@ -0,0 +1,222 @@
+
+    std::vector<std::string> linkNames(const bool scoped = false) const;
+    std::vector<std::string> jointNames(const bool scoped = false) const;
+
+    double controllerPeriod() const;
+    bool setControllerPeriod(const double period);
+
+    // ========
+    // Contacts
+    // ========
+
+    bool contactsEnabled() const;
+    bool enableContacts(const bool enable);
+
+    bool selfCollisions() const;
+    bool enableSelfCollisions(const bool enable);
+
+    std::vector<std::string> linksInContact() const;
+
+    std::vector<base::ContactData>
+    contacts(const std::vector<std::string>& linkNames = {}) const;
+
+    // ==================
+    // Vectorized Methods
+    // ==================
+
+    std::vector<double> jointForces( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointPositions( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointVelocities( //
+        const std::vector<std::string>& jointNames = {}) const;
+
+    bool setJointControlMode(const base::JointControlMode& mode,
+                             const std::vector<std::string>& jointNames = {});
+
+    std::vector<LinkPtr> links( //
+        const std::vector<std::string>& linkNames = {}) const;
+    std::vector<JointPtr> joints( //
+        const std::vector<std::string>& jointNames = {}) const;
+
+    // =========================
+    // Vectorized Target Methods
+    // =========================
+
+    bool setJointForces( //
+        const std::vector<double> forces,
+        const std::vector<std::string>& jointNames = {});
+    bool setJointPositionTargets( //
+        const std::vector<double> positions,
+        const std::vector<std::string>& jointNames = {});
+    bool setJointVelocityTargets( //
+        const std::vector<double> velocities,
+        const std::vector<std::string>& jointNames = {});
+    bool setJointAccelerationTargets( //
+        const std::vector<double> velocities,
+        const std::vector<std::string>& jointNames = {});
+
+    bool resetJointPositions( //
+        const std::vector<double> positions,
+        const std::vector<std::string>& jointNames = {});
+    bool resetJointVelocities( //
+        const std::vector<double> velocities,
+        const std::vector<std::string>& jointNames = {});
+
+    std::vector<double> jointPositionTargets( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointVelocityTargets( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointAccelerationTargets( //
+        const std::vector<std::string>& jointNames = {}) const;
+
+    // =========
+    // Base Link
+    // =========
+
+    std::string baseFrame() const;
+    bool setBaseFrame(const std::string& frameName);
+
+    bool fixedBase() const;
+    bool setAsFixedBase(const bool fixedBase);
+
+    std::array<double, 6> baseWrench() const;
```

Comment:
What is the semantics of this method? It returns the same forces returned by contactData of the canonical link but expressed in another frame?

Reply:
Woops, this method is not implemented. It is part of the constraints methods that didn't manage to get into this initial PR. I will remove it, thanks!

(likely it will be substituted by a force associated to a constraint, similarly to what PyBullet exposes)

Reply:
Addressed in https://github.com/robotology/gym-ignition/pull/158/commits/a7eb2242d7d12baedb3f3015e1a3c6e1796e1c38